stdlib/public/core/Collection.swift

```diff
@@ -24,7 +24,8 @@ public protocol Indexable {
   ///
   /// Valid indices consist of the position of every element and a
   /// "past the end" position that's not valid for use as a subscript.
-  associatedtype Index : ForwardIndex
+// TODO: swift-3-indexing-model - Index only needs to be comparable or must be comparable..?
```

Comment:
You can do that by storing an offset in the list node, provided you can give up O(1) splicing.